Refactors events page and updates resampling page (#20)
RascalSoftware/RAT-Docs: source/advanced/resampling.rst

     :width: 700
     :alt: sld and dots...

which are then converted into a set of zero roughness layers:

.. image:: ../images/advanced/layersPic.png
     :width: 700
     :alt: sld and dots...

***********************
Using Resampling in RAT
***********************

> Using resampling on a contrast in RAT is very simple. For any contrast that you want to resample, then simply set the 'resample' flag for that contrast to 'true'

Author: Should we use the back quotes on  \`\`true`` considering Python expects  True instead?

> Using resampling on a contrast in RAT is very simple. For any contrast that you want to resample, then simply set the 'resample' flag for that contrast to 'true'

Reviewer: yeah, i was thinking about that when i made the suggestion - although if python expects capital-T True then is "set the flag to 'true'" correct either? I guess it's less specific?

> Using resampling on a contrast in RAT is very simple. For any contrast that you want to resample, then simply set the 'resample' flag for that contrast to 'true'

Author: Maybe we go with "simply set the ``resample`` flag as shown below"

> Using resampling on a contrast in RAT is very simple. For any contrast that you want to resample, then simply set the 'resample' flag for that contrast to 'true'

Reviewer: sounds good to me

            controls = RAT.Controls()
            print(controls)

> * **resampleMinAngle**: The algorithm refines near the points which form, together with their left and right neighbours, a triangle with central angle smaller than the given value. So, a larger value of *resampleMinAngle* results in a finer resampling. 

Reviewer: Okay, how about:
"For each data point, the algorithm compares the two lines between that data point and its neighbouring points on either side. If the internal angle between those lines is smaller than `resampleMinAngle`, then the algorithm will refine over that point. 

In practice, this means that resampling happens for points which are significantly larger or smaller than their neighbours, and `resampleMinAngle` controls the sensitivity of this.

The algorithm terminates once there are no three adjacent points which form an angle smaller than `resampleMinAngle`."

maybe it'd make more sense if we added a picture?

> * **resampleMinAngle**: The algorithm refines near the points which form, together with their left and right neighbours, a triangle with central angle smaller than the given value. So, a larger value of *resampleMinAngle* results in a finer resampling. 

Author: I think this is better and I agree a picture would be helpful. I am still not sure about saying "points which are significantly larger or smaller than their neighbours" as far as I know points do not have size so they can't be larger or smaller. Are you still referring to the angle when you say larger or smaller?

> * **resampleMinAngle**: The algorithm refines near the points which form, together with their left and right neighbours, a triangle with central angle smaller than the given value. So, a larger value of *resampleMinAngle* results in a finer resampling. 

Reviewer: I meant 'larger or smaller' as in the function value at that point is larger or smaller. Maybe 'higher or lower' or just 'different'?
![graphcurve](https://github.com/user-attachments/assets/c2be2248-2418-46a6-8c92-41a96c264045)
Hopefully this picture explains it a bit better, with a figure caption that explains that if `resampleMinAngle` is set to 0.9, the algorithm would resample around x_1 and x_3 but not x_2.

I'm happy to add the image and explanation if you're happy with all that?

> * **resampleMinAngle**: The algorithm refines near the points which form, together with their left and right neighbours, a triangle with central angle smaller than the given value. So, a larger value of *resampleMinAngle* results in a finer resampling. 

Reviewer: done - I also noticed that the page said the algorithm resamples over angles *larger* than `resampleMinAngle`: this should be *smaller* [(see the docstring for the algorithm)](https://github.com/RascalSoftware/RAT/blob/master/targetFunctions/common/resampleLayers/adaptive.m), so I fixed all instances of that

  *resampleMinAngle* is defined as a multiple of pi (i.e. resampleMinAngle = 0.9 refines where the adjacent points form an angle greater than 0.9 * pi)

> * **resampleNPoints**: The initial number of domain points (layers). The algorithm will start from this number of layers, and refine until all the *resampleMinAngle* criteria are satisfied.

Reviewer: ```suggestion
* ``resampleNPoints``: The initial number of domain points (layers) sampled by the algorithm at the start.
```
I think the point on the `resampleMinAngle` criteria should be in the `resampleMinAngle` explanation, as above

.. note::

>         Generally, **resampleMinAngle** has a larger effect on the eventual resampling than **resampleNPoints**.

Reviewer: ```suggestion
        Generally, ``resampleMinAngle`` has a larger effect on the eventual resampling than ``resampleNPoints``.
```
